packages/next/server/app-render.tsx

```diff
@@ -1117,9 +1135,14 @@ export async function renderToHTMLOrFlight(
           // Add extra cache busting (DEV only) for https://github.com/vercel/next.js/issues/5860
           // See also https://bugs.webkit.org/show_bug.cgi?id=187726
           const cacheBustingUrlSuffix = dev ? `?ts=${Date.now()}` : ''
+          let HeadTags
+          if (rootLayoutAtThisLevel) {
+            HeadTags = collectedHeads[collectedHeads.length - 1] as any
+          }
 
           return (
             <>
+              {HeadTags ? <HeadTags /> : null}
```

Comment:
Correct me if I was wrong, but it seems that the exported `Head` can not access React Context (as their Providers are defined in the layout or page) with the current implementation.

Reply:
Correct, context is not used for this implementation and requires new experimental react APIs